drivers/dai/nxp/sai/sai.c

```diff
@@ -910,6 +919,7 @@ static struct sai_config sai_config_##inst = {					\
 	.rx_sync_mode = SAI_RX_SYNC_MODE(inst),					\
 	.tx_dline = SAI_TX_DLINE_INDEX(inst),					\
 	.rx_dline = SAI_RX_DLINE_INDEX(inst),					\
+	.pincfg = SAI_PINCTRL_DT_INST_DEV_CONFIG_GET_OPTIONAL(inst),		\
```

Comment:
I don't think these macros are needed. We already have drivers with optional pin control (IE the FlexSPI driver, since some parts have their FlexSPI wired to on-package flash). You can just check the return code of `pinctrl_apply_state`, and ignore the error if it is `-ENOENT`: https://github.com/zephyrproject-rtos/zephyr/blob/cb331c2371144f3c391856eec4417ce65cbf0957/drivers/memc/memc_mcux_flexspi.c#L264

Reply:
Thanks! Removed them